core/src/main/java/hudson/Functions.java

```diff
@@ -1128,6 +1128,43 @@ public static String getFooterURL() {
                 Jenkins.get().hasPermission(d.getRequiredGlobalConfigPagePermission()) || Jenkins.get().hasPermission(Jenkins.SYSTEM_READ)));
     }
 
+    /**
+     * Checks if the current security principal has one of the supplied permissions.
+     *
+     * @since TODO
+     */
+    public static boolean hasAnyPermission(AccessControlled ac, Permission[] permissions) {
+        if (permissions == null || permissions.length == 0) {
+            return true;
+        }
+
+        return ac.hasAnyPermission(permissions);
+    }
+
+    /**
+     * This version is so that the 'hasAnyPermission'
+     * degrades gracefully if "it" is not an {@link AccessControlled} object.
+     * Otherwise it will perform no check and that problem is hard to notice.
```

Comment:
Just throw in that case? The assumption that `object` is an ancestor isn't necessarily true. For example, in upstream cause related UI, you would show information about another job / build.

Reply:
It’s consistent with hasPermission and wasn’t working without this for the l:task it was just silently doing nothing